src/generator.rs

```diff
@@ -416,28 +405,54 @@ impl<'a> UnimplementedFunction<'a> {
             }
             TypeDefKind::Future(ty) => {
                 source.push_str("Future<");
-                self.print_optional_type(in_world, ty.as_ref(), trie, source)?;
+                self.print_optional_type(ty.as_ref(), trie, source)?;
                 source.push('>');
             }
             TypeDefKind::Stream(stream) => {
                 source.push_str("Stream<");
-                self.print_optional_type(in_world, stream.element.as_ref(), trie, source)?;
+                self.print_optional_type(stream.element.as_ref(), trie, source)?;
                 source.push_str(", ");
-                self.print_optional_type(in_world, stream.end.as_ref(), trie, source)?;
+                self.print_optional_type(stream.end.as_ref(), trie, source)?;
                 source.push('>');
             }
-            TypeDefKind::Type(ty) => self.print_type(in_world, ty, trie, source)?,
-            TypeDefKind::Handle(Handle::Own(id)) => {
-                self.print_type_id(in_world, *id, trie, source)?
-            }
+            TypeDefKind::Type(ty) => self.print_type(ty, trie, source)?,
+            TypeDefKind::Handle(Handle::Own(id)) => self.print_type_id(*id, trie, source, false)?,
             TypeDefKind::Handle(Handle::Borrow(id)) => {
-                if in_world {
+                let type_info = &self.resolve.types[*id];
```

Comment:
nit: Could we pull the body of this case out into a method called `is_export_resource` (or something)? Keeps it a little more readable IMO.

Reply:
Yep, good call

Reply:
Fixed, thanks